src/pint/toa.py

```diff
@@ -2420,11 +2453,66 @@ def merge_TOAs(TOAs_list):
     planets = [tt.planets for tt in TOAs_list]
     if len(set(planets)) > 1:
         raise TypeError(f"merge_TOAs() cannot merge. Inconsistent planets: {planets}")
-    num_cols = [len(tt.table.columns) for tt in TOAs_list]
-    if len(set(num_cols)) > 1:
-        raise TypeError(
-            f"merge_TOAs() cannot merge. Inconsistent numbers of table columns: {num_cols}"
-        )
+    # check for the presence of various computable columns
+    #   pulse_number
+    #   ssb_obs_pos, ssb_obs_vel, obs_sun_pos
+    #   tdb, tdbld
+    # if one file has these and the others don't, try to add them if strict=False
+    all_colnames = [tt.table.colnames for tt in TOAs_list]
+    has_pulse_number = np.array(
+        ["pulse_number" in colnames for colnames in all_colnames]
+    )
+    has_posvel = np.array(
```

Comment:
This logic isn't quite right if some have a few columns but not all. (For example, some have ssb_obs_pos but not ssb_obs_vel.